apps/native/app/src/screens/inbox/inbox.tsx

```diff
@@ -675,6 +844,113 @@ export const InboxScreen: NavigationFunctionComponent<{
           ) : null
         }
       />
+      {selectState && selectedItems.length ? (
+        <ActionBar
+          loading={bulkSelectActionLoading}
+          onClickStar={async () => {
+            const result = await bulkSelectActionMutation({
+              variables: {
+                input: { action: 'bookmark', documentIds: selectedItems },
+              },
+            })
+            if (result.data?.postMailActionV2?.success) {
+              // If success, update selected documents to be starred in the cache
+              selectedItems.forEach((id) => {
+                client.cache.modify({
+                  id: client.cache.identify({ __typename: 'DocumentV2', id }),
+                  fields: {
+                    bookmarked: () => true,
+                  },
+                })
+              })
+              resetSelectState()
+              showToastForBulkSelectAction({
+                variant: 'success',
+                title: intl.formatMessage({
+                  id: 'inbox.bulkSelect.starSuccess',
+                }),
+              })
+            } else {
+              showToastForBulkSelectAction({
+                variant: 'success',
+                title: intl.formatMessage({
+                  id: 'inbox.bulkSelect.starError',
+                }),
+                message: intl.formatMessage({
+                  id: 'inbox.bulkSelect.pleaseTryAgain',
+                }),
+              })
+            }
+          }}
+          onClickArchive={async () => {
+            const result = await bulkSelectActionMutation({
+              variables: {
+                input: { action: 'archive', documentIds: selectedItems },
+              },
+            })
+            if (result.data?.postMailActionV2?.success) {
+              resetSelectState()
+              res.refetch()
```

Comment:
I tried figuring out how to do this in the cache like for the other two operations but could not make it work despite following the documentation from apollo. I don't have any easy way to vizualise the cache since we upgraded to React Native 0.74 since then Flipper is not supported like it was before. Spent some time trying to set it up manually but no luck. Decided it was not worth the effort for now so refetching the data instead.